### PR TITLE
Fix handling of minus in abs, sinh, cosh, ... etc

### LINKED ISSUE #1197: Function "abs" is crashing for me
This might be a problem with my build, but the following code keeps crashing for me:

SymEngine::abs(SymEngine::add(SymEngine::mul(SymEngine::integer(-1), SymEngine::symbol("a")), SymEngine::symbol("b")));

The stacktrace looks like an infinite recursion:

stacktrace

Here's a full demo project:

[demo-abs.zip](https://github.com/symengine/symengine/files/733556/demo-abs.zip)

### PR DESCRIPTION
**Pull request by isuruf**
Fixes #1197

### CODE REVIEW

File: symengine/tests/basic/test_functions.cpp [hunk 1]
```diff
@@ -2814,6 +2844,7 @@ TEST_CASE("Abs: functions", "[functions]")
     REQUIRE(eq(*abs(neg(div(i2, i3))), *div(i2, i3)));
     REQUIRE(neq(*abs(x)->diff(x), *integer(0)));
     REQUIRE(eq(*abs(x)->diff(y), *integer(0)));
+    REQUIRE(eq(*abs(sub(x, y)), *abs(sub(y, x))));
```
Comment (isuruf) [inline symengine/tests/basic/test_functions.cpp]: @poke1024, `abs(b-a)` should be fixed by now.

Comment (poke1024) [inline symengine/tests/basic/test_functions.cpp]: Yes, fixed now. Thanks!

**Comment by isuruf** [comment]
Codecov's analysis is incorrect. The tests I've added should cover all the lines of code.

**Comment by isuruf** [comment]
@poke1024, thanks for checking.

**Comment by ranjithkumar007** [comment]
Looks good and as @isuruf  said ,the tests(14 test cases for 13 changed functions) should have covered all lines of code...Is it a bug with codecov?

**Pull request merged**